BtConnectorLib/btconnectorlib2/src/main/java/org/thaliproject/p2p/btconnectorlib/BTConnector.java

```diff
@@ -5,22 +5,19 @@
 import android.bluetooth.BluetoothDevice;
 import android.bluetooth.BluetoothSocket;
 import android.content.Context;
-import android.net.wifi.p2p.WifiP2pDevice;
 import android.net.wifi.p2p.WifiP2pManager;
-import android.os.CountDownTimer;
 import android.os.Handler;
 import android.util.Log;
-
-import java.util.Collection;
+import org.json.JSONException;
+import org.json.JSONObject;
 import java.util.List;
-import java.util.UUID;
 
 /**
  * Created by juksilve on 13.3.2015.
  */
 public class BTConnector implements BluetoothBase.BluetoothStatusChanged, WifiBase.WifiStatusCallBack, BTConnector_Discovery.DiscoveryCallback, BTConnector_BtConnection.ListenerCallback {
 
-    BTConnector that = this;
+    private final BTConnector that = this;
 
     public class WifiBtStatus{
```

Comment:
It is much cleaner and safe to make all the member variables final and require their settings to be passed into the constructor.